webdictionary/src/main/java/com/eis0/kademlianetwork/SMSKademliaListener.java

```diff
@@ -63,26 +63,21 @@ public void sendPong(SMSPeer peer){
      */
     @Override
     public void onMessageReceived(SMSMessage message) {
-        String text = message.getData();
-        SMSPeer peer = message.getPeer();
-        //Converts the code number in the message to the related enum
-        RequestTypes incomingRequest = RequestTypes
-                .values()[Integer.parseInt(text.split(" ")[0])];
-        //Array of strings containing the message fields
-        String[] splitted = text.split(" ");
+        KademliaMessage kadMessage = new KademliaMessage(message);
+        SMSPeer peer = kadMessage.peer;
+        RequestTypes incomingRequest = kadMessage.requestType;
+        KademliaId idToFind = kadMessage.idToFind;
+        SMSPeer searcher = kadMessage.searcher;
+        String key = kadMessage.key;
+        String resource = kadMessage.resource;
+        //The two values, idToFind and idFound, share the same field, depending upon the type of request
+        //the SMSKademliaListener knows which one of the two is occupying it, and process it consequently
+        KademliaId idFound = idToFind;
```

Comment:
@turcato-niccolo Now the SMSMessage received is converted into a KademliaMessage, where the fields are pre-defined. I made the choice to not make private all the variables inside the KademliaMessage, all the data are brought via SMS and are contained inside the message, which is simply converted into a more readable format; I didn't think it could be necessary to create get<Value> methods for each field, but I wait to know your opinion.